### ISSUE TITLE
Always loss nan

### ISSUE THREAD

**Issue by AnikaP** [wizyoung/YOLOv3_TensorFlow#114]
Hi,
I'm trying to train my model, but I keep getting nan for some classes.

> INFO ======> Epoch: 30, global_step: 3750.0, lr: 4.99967e-07 <======
> EVAL: Class 0: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 1: Recall: 0.1605, Precision: 0.0018, AP: 0.0013
> EVAL: Class 2: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 3: Recall: 0.0370, Precision: 0.0000, AP: 0.0000
> EVAL: Class 4: Recall: 0.3055, Precision: 0.0045, AP: 0.0014
> EVAL: Class 5: Recall: 0.2381, Precision: 0.0002, AP: 0.0001
> EVAL: Class 6: Recall: nan, Precision: 0.0000, AP: nan
> EVAL: Class 7: Recall: nan, Precision: 0.0000, AP: nan
> EVAL: Class 8: Recall: 0.0690, Precision: 0.0003, AP: 0.0000
> EVAL: Class 9: Recall: 0.4681, Precision: 0.0010, AP: 0.0006
> EVAL: Class 10: Recall: 0.8023, Precision: 0.0457, AP: 0.4857
> EVAL: Class 11: Recall: 0.1429, Precision: 0.0001, AP: 0.0000
> EVAL: Class 12: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 13: Recall: 0.0270, Precision: 0.0000, AP: 0.0000
> EVAL: Class 14: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 15: Recall: 0.2963, Precision: 0.0004, AP: 0.0003
> EVAL: Class 16: Recall: nan, Precision: 0.0000, AP: nan
> EVAL: Class 17: Recall: 0.0625, Precision: 0.0000, AP: 0.0000
> EVAL: Class 18: Recall: 0.4433, Precision: 0.0059, AP: 0.0040
> EVAL: Class 19: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 20: Recall: 0.0360, Precision: 0.0002, AP: 0.0000
> EVAL: Class 21: Recall: 0.1515, Precision: 0.0002, AP: 0.0000
> EVAL: Class 22: Recall: 0.0000, Precision: 0.0000, AP: 0.0000
> EVAL: Class 23: Recall: 0.0233, Precision: 0.0000, AP: 0.0000
> EVAL: Class 24: Recall: 0.1111, Precision: 0.0001, AP: 0.0000
> EVAL: Class 25: Recall: 0.6883, Precision: 0.0462, AP: 0.1047
> EVAL: Class 26: Recall: 0.0463, Precision: 0.0006, AP: 0.0000
> EVAL: Class 27: Recall: 0.0312, Precision: 0.0000, AP: 0.0000
> EVAL: Class 28: Recall: 0.8056, Precision: 0.0027, AP: 0.0039
> EVAL: Class 29: Recall: 0.1408, Precision: 0.0005, AP: 0.0001
> EVAL: Recall: nan, Precison: 0.0037, mAP: nan
> EVAL: loss: total: 178.79, xy: 9.25, wh: 25.00, conf: 37.39, class: 107.14

 I already tried:
* checking my anchors (I did a lot of EDA, they should work, they do for other models)
* lowering the learning rate a lot (now it's 1e-7)
* using more warm up epochs (now its 30)
* using clipping in model.py ( instead of  "box_sizes = tf.exp(box_sizes) * rescaled_anchors" it is now "box_sizes = tf.clip_by_value(tf.exp(box_sizes), 1e-9, 100) * rescaled_anchors "
* training multiply times

Any more suggestions? Maybe I should note, that I already trained another dataset successfully with my parameters. This one with the nan is pretty similar, I don't see why It should not work.
Thanks a lot for your help!

**Comment by AnikaP** [comment]
Sorry for bothering you. I had a category missing in my dataset due to a spelling error.
